src/smart-components/portfolio/share-portfolio-modal.js

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import FormRenderer from '../common/form-renderer';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { Modal } from '@patternfly/react-core';
+import { createPortfolioShareSchema } from '../../forms/portfolio-share-form.schema';
+import { addNotification } from '@red-hat-insights/insights-frontend-components/components/Notifications';
+import { fetchPortfolios } from '../../redux/actions/portfolio-actions';
+import { fetchShareInfo, sharePortfolio, unsharePortfolio } from '../../redux/actions/share-actions';
+import { fetchRbacGroups } from '../../redux/actions/rbac-actions';
+import { pipe } from 'rxjs';
+
+const SharePortfolioModal = ({
+  history: { goBack },
+  isLoading,
+  addNotification,
+  fetchPortfolios,
+  initialValues,
+  fetchShareInfo,
+  sharePortfolio,
+  fetchRbacGroups,
+  shareInfo,
+  portfolioId,
+  rbacGroups
+}) => {
+  useEffect(() => {
+    fetchShareInfo(portfolioId);
+    fetchRbacGroups();
+  }, []);
+
+  const initialShares = () => {
+    let initialGroupShareList = shareInfo.map((group) => { const groupName = group.group_name;
+      let options = permissionOptions.find(perm => (perm.value === group.permissions.sort().join(',')));
+      return { [groupName]: options ? options.value : 'Unknown' };});
+    let initialShareList = initialGroupShareList.reduce((acc, curr) => ({ ...acc, ...curr }), {});
+
+    return initialShareList;
+  };
+
+  const onSubmit = data =>
+  {
+    let sharePromises = [];
+    if (data.group_uuid && data.permissions) {
+      sharePromises.push(sharePortfolio(data));
+    }
+
+    shareInfo.map(share => {
```

Comment:
`map` is supposed to be used when you want to use create new array from input array. For this use case i would prefer `forEach`